docs/achitecture_and_design/api_design.md

```diff
@@ -0,0 +1,121 @@
+# API DESIGN
+
+The following are the RESTful endpoints needed to support basic chat and follow-up creation, retrieval, updating, and deletion.
+
+## Chats
+
+### Retrieve individual chat
+`GET /api/chats/<id>`
+```json
+{
+    "id": 2,
+    "user": {
+        "id": 1,
+        "name": "..."
+    },
+    "title": "...",
+    "description": "...",
+    "date": "...",
+    "follow_ups_url": "api/chats/<id>/followups",
+    "follow_up_count": 5
+}
+```
+
+### Retrieve a user's chats
+`GET /api/user/2/chats`
+```json
+{
+    "count": 123,
+    "next": "...url to next page of results...",
+    "previous": "...url to next previous of results...",
+    "chats": [
+        {
+            "id": 1,
+            "user": {
+                "id": 1,
+                "name": "..."
+            },
+            "title": "...",
+            "description": "...",
+            "date": "...",
+            "follow_ups_url": "api/chats/<id>/followups",
+            "follow_up_count": 5
+        },
+        ...
+    ]
+}
+```
+
+### Retrive all chats
+`GET /api/chats`
+(Same data returned as User's Chats)
```

Comment:
these are the chats for the currently logged in user?

Reply:
In our current product, all chats are public and `/api/chats` will simply return all chats that exist. Obviously this will be a big number at some point and we'll do away with this. But as stated here, it represents a replacement for [our forum](https://groups.google.com/forum/#!forum/penny-university) which is a list of all chats.